### PR TITLE
Initial support to defer DRM ModeSet

### PR DESCRIPTION
**Pull request by blindmatrix**
## What does the pull request do?
This will allow applications to defer the actual modeset (drmModeSetCrtc) until the first frame has been generated.

## What is the current behavior?
When starting Avalonia with the DRM backend you'll get a black screen for a few seconds before the UI appear. This happens because the screen is cleared and configured when the output backend is created and on slower hardware result in a blank screen for up to 8 seconds. When using the framebuffer output instead you can leave a splash screen (or some sort of animation) running until the UI is ready.

## What is the updated/expected behavior with this PR?
This will (optionally) wait with the screen blanking until there's actual content to display.

## How was the solution implemented (if it's not obvious)?
If you opt to defer the mode set the output will do the mode set instead of a page flip on your first presentation.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Breaking changes
None as this behavior is opt-in with this patch and handled by using argument defaults.

### CODE REVIEW

File: src/Linux/Avalonia.LinuxFramebuffer/Output/DrmOutput.cs [hunk 1]
```diff
@@ -203,47 +214,66 @@ public RenderSession(DrmOutput parent, IDisposable clearContext)
...
-                        DrmEventPageFlipHandlerDelegate flipCb =
-                            (int fd, uint sequence, uint tv_sec, uint tv_usec, void* user_data) =>
-                            {
-                                waitingForFlip = false;
-                            };
-                        var cbHandle = GCHandle.Alloc(flipCb);
-                        var ctx = new DrmEventContext
+                private void DoPageFlipOrModeSet(IntPtr nextBo)
+                {
+                    if (_parent._deferedModesetActivator is not null)
```
Comment (Takoooooo) [inline src/Linux/Avalonia.LinuxFramebuffer/Output/DrmOutput.cs:238]: I think this feature is not supported on our SDK. Please update

Comment (maxkatz6) [reply src/Linux/Avalonia.LinuxFramebuffer/Output/DrmOutput.cs:238]: Downgrade code *
Our CI build doesn't work with latest SDK yet.

**Comment by kekekeks** [comment]
Haven't checked the code yet, but wouldn't providing a splash screen image be a better solution that deferring an operation that can potentially fail later?

**Comment by blindmatrix** [comment]
> Haven't checked the code yet, but wouldn't providing a splash screen image be a better solution that deferring an operation that can potentially fail later?

Yes, but if you're running your application with DRM I'm assuming most instances would be to have it embedded on hardware, which you'd probably already know that it will work. But yes a splash screen would be better.

Our actual problem is that it take about 15 seconds during boot to present the first screen. The hardware is quite limited (imx8mm) and all cores compete with starting the system (we start our frontend during sysinit.target to get it going early). Our frontend (currently) doesn't really do anything except present some buttons, but most of the start time (I think) is spent initializing FluentTheme. We want to run some sort of animated splash while we're waiting so to have a splash screen presented with the help of Avalonia would require it to do so before parsing the themes and such. (Starting the UI once everything else is running take a few seconds)

Having a static image displayed from the constructor of DrmOutput (of the framebuffer alternative) would be quite simple, but we want it animated.

Long story short, it would be great to have a splash screen managed by the framework instead, but without waiting on most of it's internal initialization.

Any ideas?

**Comment by kekekeks** [comment]
I guess we could provide a some API with some thread-safe drawing APIs (only images, immutable brushes and already prepared FormattedText instances) for drawing an animated splash screen while the main thread is busy loading.

**Comment by blindmatrix** [comment]
That would be an excellent solution!

**Comment by jmacato** [comment]
Closing due to inactivity, ping me if you want to work on this again.
